src/biome/text/tokenizer.py

```diff
@@ -20,61 +21,47 @@ class Tokenizer(FromParams):
     
     Parameters
     ----------
-    lang
-        The [spaCy model used](https://spacy.io/api/tokenizer) for tokenization is language dependent.
-        For optimal performance, specify the language of your input data (default: "en").
-    max_sequence_length: `int`
-        Maximum length in characters for input texts truncated with `[:max_sequence_length]` after `TextCleaning`.
-    max_nr_of_sentences: `int`
-        Maximum number of sentences to keep when using `segment_sentences` truncated with `[:max_sequence_length]`.
-    text_cleaning: `Optional[TextCleaning]`
-        A `TextCleaning` configuration with pre-processing rules for cleaning up and transforming raw input text.
-    segment_sentences:  `Union[bool, SentenceSplitter]`
-        Whether to segment input texts in to sentences using the default `SentenceSplitter` or a given splitter.
-    start_tokens: `Optional[List[str]]`
-        A list of token strings to the sequence before tokenized input text.
-    end_tokens: `Optional[List[str]]`
-        A list of token strings to the sequence after tokenized input text.
+    config
+        A `TokenizerConfiguration` object
     """
-
-    # TODO(dcfidalgo): Check if it is possible to pass in a `TokenizerConfiguration` to avoid duplication of arguments
-
     def __init__(
         self,
-        lang: str = "en",
-        max_sequence_length: int = None,
-        max_nr_of_sentences: int = None,
-        text_cleaning: Optional[TextCleaning] = None,
-        segment_sentences: Union[bool, SentenceSplitter] = False,
-        start_tokens: Optional[List[str]] = None,
-        end_tokens: Optional[List[str]] = None,
+        config: TokenizerConfiguration,
     ):
-        self._fetch_spacy_model(lang)
+        self.lang = config.lang
+        self._fetch_spacy_model()
+
+        if config.segment_sentences is True:
+            self.segment_sentences = SpacySentenceSplitter(language=self.lang, rule_based=True)
+        elif config.segment_sentences is False:
+            self.segment_sentences = None
+        else:
+            self.segment_sentences = SentenceSplitter.from_params(Params(config.segment_sentences))
```

Comment:
You must pass to allennlp `Params` deep copy to avoid to lost original configuration (required for `pipeline.config` property)

allennlp `from_params` read dictionary params using `pop` method and checks an empty dictionary at the end

Reply:
done, forgot about this behaviour!